File: Assignment 01/S_Max_Split.py
s = input()
l =0
r = 0
idx = 0
ans =""
arr =[]
for i in s:
    ans += i
    if i == 'L':
        l=l+1
    else :
        r = r+1
    if l==r:
        arr.append(ans)
        ans =""
        idx=idx+1
        l = 0
        r = 0
print(idx)
for char in arr:
    print(char)

File: Assignment 01/S_Max_Split.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
    string s;
    cin>>s;
    int l=0,r=0,idx=0;
    string arr[1000];
    string ans ="";
    for(int i=0;i<s.size();i++)
    {
        if(s[i]=='L')l++;
        else r++;
        ans.push_back(s[i]);
        if(l==r)
        {
            arr[idx] = ans;
            ans = "";
            idx++;
            l=0;
            r=0;
        }
    } 
    cout<<idx<<endl;
    for(int i=0;i<idx;i++)
    {
        cout<<arr[i]<<endl;
    }
    return 0;
}